src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -8525,12 +8525,12 @@ s32 func_80843E64(PlayState* play, Player* this) {
     return 0;
 }
 
-void func_8084409C(PlayState* play, Player* this, f32 speedXZ, f32 velocityY) {
+void func_8084409C(PlayState* play, Player* this, f32 speed, f32 velocityY) {
```

Comment:
I changed it cause I felt that it being matched with the instance var felt better. but yeah it also taking y velocity is a good point.
Semi related, when we rename `linearVelocity` we will have to make the same kind of decision. `speedTarget` for parity with `actor.speed` or `speedXZTarget` and let it be slightly different

Reply:
>  Semi related, when we rename `linearVelocity` we will have to make the same kind of decision. `speedTarget` for parity with `actor.speed` or `speedXZTarget` and let it be slightly different

I can hardly wait

> also this is more than just pots, its pretty much any thrown actor afaik

Yeah I just took those as examples
I also tried looking at ruto but monkaUnk
and couldn't think of others

Reply:
yea there are a few others
a small part of me still thinks it should maybe be more general, since even if the current throwable actors in the game use gravity movement, it doesnt mean all will (thinking modding scenarios). But yea still the y vel arg is a thing

Reply:
> it doesnt mean all will (thinking modding scenarios)

Yeah at that point I think you would either change this player function `func_8084409C` to account for that
Or in the XYZ-speed-using actor, """convert""" the speed from XZ to XYZ like `speed = sqrt(sq(speed) + sq(velocity.y))`

Reply:
yeah, fair